### ISSUE TITLE
How to make this repo work for only inference

### ISSUE THREAD

**Issue by HassanBinAli** [Daraan/CropAndWeedDetection#4]
Hey! Thanks for your work, I want to only use this repo to take inference on some data. I believe I found the script, but I don't think that you provided weights/checkpoints of your training. Your help in this regard would be appreciated.

**Comment by Daraan** [comment]
Hey,

I should still have the checkpoints.

I'll look for them & a way to share them; sadly I just think the whole collection was around 40GB.
Feel free to remind me at the end of next week if I did not do it by then.

**Comment by HassanBinAli** [comment]
Thanks for your response! So you may share that checkpoint only which you think is the best. This way it would be easy for you as well as for those who are going to use this repo.

**Comment by HassanBinAli** [comment]
Hey! did you find time to finalize the weights?

**Comment by Daraan** [comment]
Hey, I am still looking into it.
Sadly the weights were moved/deleted in some cleanup process as our space restrictions were changed quite drastically. It could be that on the PC I was working that time is a local copy but haven't been able to verify that.

---

I am looking into it currently and should be able to provide something in the next days.

I provided a simple model to test the setup that I did not commit yet.

**Comment by Daraan** [comment]
Hi Hassan I just uploaded three models.

The two large ones (e6 and x) are stored through [git lfs](https://docs.github.com/articles/versioning-large-files/)
You either need git lfs to download them or use the release where they should be included https://github.com/Daraan/CropAndWeedDetection/releases/tag/Inference-models-(1) 

---

## Models

### e6 model (1280px): 

Stored in Git LFS

* validation mAP (30 Epochs): 0.763
* test mAP (40 Epochs) - needs reevaluation at 30!:

```md
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       Test metric             DataLoader 0
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
        test_loss            3.747124433517456
        test_map            0.6489368081092834
       test_map_50          0.7916330695152283
       test_map_75           0.698529839515686
      test_map_Bean         0.6892199516296387
     test_map_Maize         0.7442483305931091
      test_map_Pea          0.5929533243179321
     test_map_Potato        0.6119544506072998
    test_map_Pumpkin        0.8598697185516357
      test_map_Soy          0.46878740191459656
   test_map_Sugar beet      0.7401812076568604
   test_map_Sunflower       0.7893890738487244
      test_map_Weed         0.3438277244567871
     test_map_small         0.6489368081092834
    test_mar100_Bean        0.7127273082733154
    test_mar100_Maize       0.7801679372787476
     test_mar100_Pea        0.6573913097381592
   test_mar100_Potato       0.6965516805648804
   test_mar100_Pumpkin      0.8676470518112183
     test_mar100_Soy         0.515816330909729
 test_mar100_Sugar beet     0.7667056918144226
  test_mar100_Sunflower     0.8092250823974609
    test_mar100_Weed        0.3837168514728546
       test_mar_1           0.22914600372314453
       test_mar_10          0.6714540123939514
      test_mar_100          0.6877721548080444
     test_mar_small         0.6877721548080444
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
```


### x model (640px): 

Stored in Git LFS

Trained with gradient accumulation over two epochs of batch size 16:

* validation mAP (48 Epochs): 0.789
* test mAP (50 Epochs) - needs reevaluation at 48:

```md
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       Test metric             DataLoader 0
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
        test_loss            4.777414798736572
        test_map             0.674770176410675
       test_map_50          0.8385519981384277
       test_map_75          0.7259414196014404
      test_map_Bean         0.7226269245147705
     test_map_Maize         0.7365819215774536
      test_map_Pea          0.6495985984802246
     test_map_Potato        0.4847957193851471
    test_map_Pumpkin        0.9078474640846252
      test_map_Soy          0.5573299527168274
   test_map_Sugar beet       0.784260094165802
   test_map_Sunflower       0.8126767873764038
      test_map_Weed          0.417214035987854
     test_map_small          0.674770176410675
    test_mar100_Bean        0.7587879300117493
    test_mar100_Maize       0.7741867303848267
     test_mar100_Pea        0.7070175409317017
   test_mar100_Potato       0.5999999642372131
   test_mar100_Pumpkin      0.9205881953239441
     test_mar100_Soy        0.6239796280860901
 test_mar100_Sugar beet      0.819741427898407
  test_mar100_Sunflower     0.8416973948478699
    test_mar100_Weed        0.4872349798679352
       test_mar_1           0.22704993188381195
       test_mar_10          0.6831567287445068
      test_mar_100          0.7259149551391602
     test_mar_small         0.7259149551391602
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

```

### tiny (640px):

* validation mAP (38 Epochs): 0.592
* test mAP (40 Epochs) - needs reevaluation at 38:

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
       Test metric             DataLoader 0
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
        test_loss           11.871225357055664
        test_map            0.48474761843681335
       test_map_50           0.710361897945404
       test_map_75          0.5239822268486023
      test_map_Crop         0.6333568096160889
      test_map_Weed         0.3361384868621826
     test_map_small         0.48474761843681335
    test_mar100_Crop        0.7037509679794312
    test_mar100_Weed        0.42132312059402466
       test_mar_1           0.13459208607673645
       test_mar_10          0.48484864830970764
      test_mar_100          0.5625370144844055
     test_mar_small         0.5625370144844055
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

**Comment by HassanBinAli** [comment]
Thank you Daraan!
